### DISCUSSION TITLE
How to create Avalonia controls that "float" aligned bottom right above other controls?

### DISCUSSION THREAD

**Discussion by uxmal** [AvaloniaUI/Avalonia#15638] [Q&A] [answered]
I'm repeating my StackOverflow question here, in the hope that I might get a "Avalonic" reply:

https://stackoverflow.com/questions/78440273/

I'm porting a Windows Forms view that looks like this:

![image](https://github.com/AvaloniaUI/Avalonia/assets/10154465/59bf96f0-2c10-4462-8bfe-3d140e2a1ca6)

The `Loading... (56%)` is a `Label` control that has been added to the `Form` with the `Anchor` property set to `Bottom, Right`, so it will stay in the bottom right as the form is resized. Its Z-order is such that it always appears in front of the "Lorem ipsum" text box in the background, which is docked to fill the form's area. Once loading is completed the label is hidden.

How do I port this to Avalonia? I was looking at the Canvas class which allows full freedom of placement, but which puts a burden on the programmer to update the `Canvas.Left` etc properties accurately. I'd like the `Label` control to assume its "natural" size, and then have it stick to the bottom right. How can I achieve this?

**Comment by jp2masa** [comment]
Did you try using `Canvas.Right="0" Canvas.Bottom="0"`? Alternatively, you can use a `Grid` with 2 rows and set `Grid.RowSpan="2"` in the text box.

**Comment by MikeCodesDotNET** [comment, marked as answer]
The following XAML will produce a view similar to what you're looking for. 

```
<Panel>
        <TextBox Text="{Binding Greeting}"
                   HorizontalAlignment="Stretch" 
                   VerticalAlignment="Stretch"
                   TextWrapping="Wrap"/>
        
        <TextBlock Text="Loading... (56%)" 
                   Background="Blue"
                   Foreground="White"
                   Padding="10"
                   VerticalAlignment="Bottom"
                   HorizontalAlignment="Right"/>
 </Panel>
```

![CleanShot 2024-05-07 at 09 03 08@2x](https://github.com/AvaloniaUI/Avalonia/assets/552074/496751bc-7bfb-4094-88e9-17a4a7a004cd)

**Reply by uxmal** [reply]
@MikeCodesDotNET awesome! Your XAML does exactly what I wanted. I had the following horrible XAML:

```xml
	<DockPanel Name="dp">
		<TextBlock
			Background="#E0D0D0"
			TextWrapping="Wrap"
                        Text="{Binding LoremIpsum" />
		<Canvas Width="0">
			<Label Name="l"
				Canvas.Bottom="0"
				Canvas.Right="0"
				Background="Blue" Foreground="White">Loading... (56%)</Label>
		</Canvas>
	</DockPanel>
```
This leaves an unfilled column on the right, which flickers when resized:

![image](https://github.com/AvaloniaUI/Avalonia/assets/10154465/da43772b-a0ea-46c9-9569-534a1460d16f)
